inst/include/Rcpp/sugar/matrix/as_vector.h

```diff
@@ -30,8 +30,8 @@ inline Rcpp::Vector<RTYPE>
 as_vector__impl( MatrixBase<RTYPE,NA,T>& t, Rcpp::traits::false_type ){
     T& ref = t.get_ref() ;
     int nc = ref.ncol(), nr = ref.nrow() ;
-    Vector<RTYPE> out (nr*nc) ;
-    int k =0;
+    Vector<RTYPE> out (((R_xlen_t)nr) * nc) ;
```

Comment:
We usually prefer C++-style casts (`static_cast<R_xlen_t>(nr)`) over C-style casts.